java/configuration/src/main/java/sleeper/configuration/utils/BucketUtils.java

```diff
@@ -27,12 +27,22 @@
 import java.util.List;
 import java.util.function.Predicate;
 
+/**
+ * Util for interacting with S3 buckets.
+ */
 public class BucketUtils {
     private static final Logger LOGGER = LoggerFactory.getLogger(BucketUtils.class);
 
     private BucketUtils() {
     }
 
+    /**
+     * Checks if an s3 bucket exists.
+     *
+     * @param  s3         the S3client to interact with AWS.
+     * @param  bucketName the string name of the bucket to check.
+     * @return            the boolean for if the bucket exists.
```

Comment:
These `@param` and `@return` tags are phrases rather than sentences, so should not end with a full stop.

The type in "the S3client" is capitalised oddly. It's not capitalised as a word in a sentence, or as the Java class name. I'd probably just make it "the client".

There are similar problems for the other methods in this class as well.

See the Oracle style guide:

https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#@param